ORBIT/manager.py

```diff
@@ -750,14 +784,162 @@ def project_logs(self):
+
+        _expense_logs = self._filter_logs(keys=["cost", "time"])
+        expenses = np.array(
+            _expense_logs, dtype=[("cost", "f8"), ("time", "i4")]
+        )
+        dig = np.digitize(expenses["time"], self.month_bins)
+
+        monthly = {}
+        for i in range(1, lifetime * 12):
+            monthly[i] = sum(expenses["cost"][dig == i]) + opex[i]
+
+        return monthly
+
+    @property
+    def monthly_opex(self):
+        """Returns the monthly OpEx expenditures based on project size."""
+
+        rate = self.config.get("opex_rate", 150)
+        lifetime = self.config.get("project_lifetime", 25)
+
+        try:
+            times, turbines = self.progress.energize_points
+            dig = list(np.digitize(times, self.month_bins))
+
+        except ValueError:
+            return {i: 0.0 for i in range(1, lifetime * 12)}
+
+        opex = {}
+        for i in range(1, lifetime * 12):
+            generating_strings = len([t for t in dig if i >= t])
+            generating_turbines = sum(turbines[:generating_strings])
+
+            opex[i] = (
+                generating_turbines * self.turbine_rating * rate * 1000 / 12
+            )
+
+        return opex
+
+    @property
+    def monthly_revenue(self):
+        """Returns the monthly revenue based on when array system strings can
+        be energized, eg. 'self.progress.energize_points'."""
+
+        ncf = self.config.get("ncf", 0.4)
+        price = self.config.get("offtake_price", 80)
+        lifetime = self.config.get("project_lifetime", 25)
+
+        times, turbines = self.progress.energize_points
+        dig = list(np.digitize(times, self.month_bins))
+
+        revenue = {}
+        for i in range(1, lifetime * 12):
+            generating_strings = len([t for t in dig if i >= t])
+            generating_turbines = sum(turbines[:generating_strings])
+            production = (
+                generating_turbines * self.turbine_rating * ncf * 730
+            )  # MWh
+            revenue[i] = production * price
+
+        return revenue
+
+    @property
+    def cash_flow(self):
+        """Returns the net cash flow based on `self.monthly_expenses` and
+        `self.monthly_revenue`."""
+
+        try:
+            revenue = self.monthly_revenue
+
+        except ValueError:
+            revenue = {}
+
+        expenses = self.monthly_expenses
+        return {
+            i: revenue.get(i, 0.0) - expenses.get(i, 0.0)
+            for i in range(1, max([*revenue.keys(), *expenses.keys()]) + 1)
+        }
+
+    @property
+    def npv(self):
+        """Returns the net present value of the project based on
+        `self.cash_flow`."""
+
+        dr = self.config.get("discount_rate", 0.025)
+        pr = (1 + dr) ** (1 / 12) - 1
+
+        cash_flow = self.cash_flow
+        _npv = [
+            (cash_flow[i] / (1 + pr) ** (i))
+            for i in range(1, max(cash_flow.keys()) + 1)
+        ]
+
+        return sum(_npv)
```

Comment:
This should return npv = (overnight capital cost) - (sum of discounted monthly cash flows)

Reply:
Updated.

Reply:
It was updated but I guess it locks the snippet above at where you made the comment.

Reply:
Updated method: https://github.com/WISDEM/ORBIT/blob/fe761c71fab311337b93bc97a24ace22e870aa01/ORBIT/manager.py#L887-L901